src/test/java/de/tum/in/www1/artemis/connector/AthenaRequestMockProvider.java

```diff
@@ -0,0 +1,190 @@
+package de.tum.in.www1.artemis.connector;
+
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.*;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.*;
+
+import java.net.SocketTimeoutException;
+
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.test.web.client.ExpectedCount;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.test.web.client.RequestMatcher;
+import org.springframework.test.web.client.ResponseActions;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+@Component
+@Profile("athena")
+public class AthenaRequestMockProvider {
+
+    private final RestTemplate restTemplate;
+
+    private MockRestServiceServer mockServer;
+
+    private final RestTemplate shortTimeoutRestTemplate;
+
+    private MockRestServiceServer mockServerShortTimeout;
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
```

Comment:
can be final, see my remarks in ```AthenaFeedbackSuggestionsService ```

Reply:
Let's discuss all instances of this here: https://github.com/ls1intum/Artemis/pull/6861#discussion_r1277559597